engine/src/engine.rs

```diff
@@ -844,6 +858,19 @@ pub fn submit<I: IO + Copy, E: Env, P: PromiseHandler>(
             .map_err(|_e| EngineErrorKind::InvalidSignature)?
     };
 
+    let fixed_gas_cost = silo::get_fixed_gas_cost(&io);
+    let transaction = if fixed_gas_cost.is_some() {
```

Comment:
@aleksuss why do we push gas limit to the max in the case of a Silo?

Reply:
To exclude a `OutOfGas` error.

Reply:
Can you add this as a comment?